cmd/ci-secret-generator/main_test.go

```diff
@@ -435,3 +437,319 @@ func TestValidateConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateSecretsWithGSMIndex(t *testing.T) {
+	t.Parallel()
```

Comment:
Beware that child tests spawned by a "parallel" parent don't run in parallel each other. 
An example is worth more:
```golang
func Test1(t *testing.T) {
  t.Parallel()
  t.Run("subtest1", func(t *testing.T) {
    // ...
  })
}

func Test2(t *testing.T) {
  t.Parallel()
  t.Run("subtest2", func(t *testing.T) {
    // ...
  })
}
```
`subtest1` and `subtest2` **do** run in parallel but:
```golang
func Test1(t *testing.T) {
  t.Parallel()
  for i := range 2 {
    t.Run(fmt.Sprintf("subtest1.%d", i), func(t *testing.T){
          // ...
    })
  }
}
```
`subtest1.1` and `subtest1.2` don't. In order to achieve that you have to move `t.Parallel()` within `t.Run(...)`, see [here](https://github.com/golang/go/blob/e8ed85d6c22d9530523315c3048ed0455d205bff/src/testing/testing.go#L1824-L1826).